ports/vcpkg-cmake/vcpkg_cmake_configure.cmake

```
@@ -159,71 +159,18 @@ function(vcpkg_cmake_configure)
     if(DEFINED arg_GENERATOR)
         set(generator "${arg_GENERATOR}")
     elseif(arg_WINDOWS_USE_MSBUILD OR NOT ninja_can_be_used)
-        if(VCPKG_PLATFORM_TOOLSET STREQUAL "v120")
-            if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-                set(generator "Visual Studio 12 2013")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-                set(generator "Visual Studio 12 2013 Win64")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
-                set(generator "Visual Studio 12 2013 ARM")
-            endif()
-        elseif(VCPKG_PLATFORM_TOOLSET STREQUAL "v140")
-            if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-                set(generator "Visual Studio 14 2015")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-                set(generator "Visual Studio 14 2015 Win64")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
-                set(generator "Visual Studio 14 2015 ARM")
-            endif()
-        elseif(VCPKG_PLATFORM_TOOLSET STREQUAL "v141")
-            if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-                set(generator "Visual Studio 15 2017")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-                set(generator "Visual Studio 15 2017 Win64")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
-                set(generator "Visual Studio 15 2017 ARM")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
-                set(generator "Visual Studio 15 2017")
-                set(arch "ARM64")
-            endif()
-        elseif(VCPKG_PLATFORM_TOOLSET STREQUAL "v142")
-            set(generator "Visual Studio 16 2019")
-            if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-                set(arch "Win32")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-                set(arch "x64")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
-                set(arch "ARM")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
-                set(arch "ARM64")
-            else()
-                set(generator)
-            endif()
-        elseif(VCPKG_PLATFORM_TOOLSET STREQUAL "v143")
-            set(generator "Visual Studio 17 2022")
-            if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-                set(arch "Win32")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-                set(arch "x64")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
-                set(arch "ARM")
-            elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
-                set(arch "ARM64")
-            else()
-                set(generator)
-            endif()
-        endif()
-    else()
-        set(generator "Ninja")
+        set(generator "")
+        set(arch "")
+        z_vcpkg_get_visual_studio_generator(generator arch)
     endif()
 
-    if(NOT DEFINED generator)
+    if(NOT DEFINED generator  OR generator STREQUAL "")
```

```suggestion
    if(NOT DEFINED generator OR generator STREQUAL "")
```

Can you remove the double whitespace?

---

IMO it should be just `IF(NOT generator)`. But there is a different change in #21507 anyways.

---

I don't feel like forcing CI to do a big rebuild due to just a WS change. Especially if the section will be changed by other PRs anyway.